File: src/main/java/org/jukeboxmc/world/generator/EmptyGenerator.java
package org.jukeboxmc.world.generator;

import org.jukeboxmc.math.Vector;
import org.jukeboxmc.world.chunk.Chunk;
import org.jukeboxmc.world.chunk.manager.PopulationChunkManager;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public class EmptyGenerator extends Generator {

    @Override
    public void generate( Chunk chunk, int chunkX, int chunkZ ) {

    }

    @Override
    public void populate( PopulationChunkManager manager, int chunkX, int chunkZr ) {

    }

    @Override
    public void finish( PopulationChunkManager manager, int chunkX, int chunkZ ) {

    }

    @Override
    public Vector getSpawnLocation() {
        return new Vector( 0, 64, 0 );
    }
}

File: JukeboxMC-API/src/main/kotlin/org/jukeboxmc/api/world/generator/EmptyGenerator.kt
package org.jukeboxmc.api.world.generator

import org.jukeboxmc.api.math.Vector
import org.jukeboxmc.api.world.chunk.Chunk

class EmptyGenerator: Generator() {

    override fun generate(chunk: Chunk, chunkX: Int, chunkZ: Int) {

    }

    override fun populate(manager: PopulationChunkManager, chunkX: Int, chunkZ: Int) {

    }

    override fun finish(manager: PopulationChunkManager, chunkX: Int, chunkZ: Int) {

    }

    override fun spawnLocation(): Vector {
        return Vector(0, 64, 0)
    }
}